tests/framework/clients/rancher/provisioning/cluster.go

```diff
@@ -17,5 +21,23 @@ func (c *Cluster) Create(ctx context.Context, cluster *apisV1.Cluster, opts meta
 
 		return err
 	})
+	watchInterface, err := c.Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + cluster.GetName(),
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	c.ts.RegisterWaitFunc(func() error {
```

Comment:
see comments in session.go

Reply:
Same here, it's not actually waiting, it's just registering it's wait function.